.github/workflows/upstream-dev-ci.yaml

```
@@ -14,7 +14,11 @@ jobs:
   upstream-dev:
     name: upstream-dev
     runs-on: ubuntu-latest
-    if: github.repository == 'pydata/xarray'
+    if: |
+      contains(github.event.head_commit.message, 'test-upstream')
+      || (github.repository == 'pydata/xarray' 
+          && github.event_name != 'pull_request' 
+          && !contains(github.event.head_commit.message, 'skip-ci'))
```

not sure if I understand this correctly, but I think the CI should never be run on a different repository, and neither the scheduled CI nor `workflow_dispatch` should ever be skipped using a commit message (especially since squash-merging concatenates the first comment of a PR and all of its commit messages):
```suggestion
      github.repository == 'pydata/xarray'
      && (
          (github.event_name == 'scheduled' || github.event_name == 'workflow_dispatch')
          || (
              contains(github.event.head_commit.message, 'test-upstream')
              && !contains(github.event.head_commit.message, 'skip-ci')
          )
      )
```